### DISCUSSION TITLE
STL Forecast

### DISCUSSION THREAD

**Discussion by eyalshafran** [alan-turing-institute/sktime#1177] [Ideas]
As far as I can tell there is no STL Forecaster implemented yet. I was thinking about the best way to do so. 
There are two options I can think off:
1. Create an STL transformation to remove the seasonality then use a TransformedTargetForecaster and predict the trend with any forecasters. The inverse transform works differently in-sample vs out-of-sample and this will have to be taken into account in the transformation.
2. Create a forecaster class that does everything (similar to statsmodels). The down side is that the forecaster (e.g. autoETS, ARIMA) will have to be selected as a parameter in the class and supported. 

I think option 1 is better.

Any thoughts? Is someone already working on this?

**Comment by mloning** [comment]
Hi @eyalshafran - thanks for the feature idea! There is ongoing work on this here: https://github.com/alan-turing-institute/sktime/pull/933 

@thayeylolu do you think you need help with your PR in case @eyalshafran would want to contribute?

**Reply by fkiraly** [reply]
I think there was some problem with the `update` method of the `TransformedTargetForecaster` that never got resolved. At least that's my latest on this. Perhaps @aiwalter can comment?
